src/main/scala/com/github/codelionx/distod/actors/Master.scala

```diff
@@ -219,19 +231,124 @@ class Master(context: ActorContext[Command], stash: StashBuffer[Command], localP
   }
 
   private def generateLevel1(
-      attributes: Seq[Int], partitions: Array[FullPartition]
+      attributes: Seq[Int],
+      partitions: Array[FullPartition]
   ): Map[CandidateSet, CandidateState] = {
     val L1candidates = attributes.map(columnId => CandidateSet.from(columnId))
     val L1candidateState = L1candidates.map { candidate =>
       candidate -> CandidateState(
-        splitCandidates = attributes,
+        splitCandidates = CandidateSet.fromSpecific(attributes),
         swapCandidates = Seq.empty,
-        isValid = true
+        swapChecked = true // we do not need to check for swaps in level 1 (single attribute nodes)
       )
     }
     L1candidates.zipWithIndex.foreach { case (candidate, index) =>
       partitionManager ! InsertPartition(candidate, partitions(index))
     }
     L1candidateState.toMap
   }
+
+  private def generateNewCandidates(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4593697)